calculator/main.py

```diff
@@ -25,9 +26,21 @@ def perform_operation(calc, operation, params):
     :param operation: String with the operation name. ie: 'add'
     :param params: Tuple containing the list of nums to operate with.
                    ie: (1, 2, 3, 4.5, -2)
+                   
     """
-    pass
-
+    if operation not in calc['operations'].keys():
+        raise InvalidOperation("Given operation is invalid.")
+
+ 
+    for element in params:
+        if type(element) == str:
+            raise InvalidParams("Given params are invalid.")
+    
+    now = datetime.now() 
+    time = "{:02d}-{:02d}-{:02d} {:02d}:{:02d}:{:02d}".format(now.year, now.month, now.day, now.hour, now.minute, now.second)
```

Comment:
This is going to be helpful: https://docs.python.org/2/library/datetime.html#datetime.date.strftime

Basically it's going to be:

```python
datetime.now().strftime('%Y-%m-%d %H:%m:%S')
```